cypress/support/commands.ts

```diff
@@ -0,0 +1,168 @@
+       */
+      beforeNationalTests(pageName: string): void;
+      /**
+       * Fixture loading and page navigation for regional page tests
+       *
+       * @param pageName
+       * @param vrcode
+       */
+      beforeRegionTests(pageName: string, vrcode?: string): void;
+      /**
+       * Ignores any errors coming out of the ResizeObserver
+       */
+      swallowResizeObserverError(): void;
+      /**
+       * Checks a list of values against their associated of data-cy IDs
+       *
+       * @param kpiTestInfo
+       */
+      checkKpiValues(kpiTestInfo: Record<string, string | string[]>): void;
+      /**
+       * Checks if there is only ONE <h1> element on the page
+       *
+       */
+      checkHeadings(): void;
+    }
+  }
+}
+
+Cypress.Commands.add('checkHeadings', () => {
+  const headings = Cypress.$('h1');
+  expect(headings.length).to.equal(
+    1,
+    'More than one <H1> element was found on the page, only one is allowed. This might be because there are multiple <ContentHeader> components on the page that all have a category prop assigned. This prop is rendered as an <H1> element. Remove one of those props to fix this error.'
+  );
+});
+
+Cypress.Commands.add('beforeGeneralTests', (pageName: string) => {
+  cy.visit(`/${pageName}`);
+
+  cy.checkHeadings();
+});
+
+Cypress.Commands.add(
+  'beforeMunicipalTests',
+  (pageName: string, gmcode = 'GM0363') => {
+    cy.swallowResizeObserverError();
+
+    cy.fixture<Municipal>(`${gmcode}.json`)
+      .as('municipalData')
+      .visit(`/gemeente/${gmcode}/${pageName}`);
+
+    cy.checkHeadings();
+  }
+);
+
+Cypress.Commands.add('beforeNationalTests', (pageName: string) => {
+  cy.swallowResizeObserverError();
+
+  cy.fixture<National>('NL.json')
+    .as('nationalData')
+    .visit(`/landelijk/${pageName}`);
+
+  cy.checkHeadings();
+});
+
+Cypress.Commands.add(
+  'beforeRegionTests',
+  (pageName: string, vrcode = 'VR13') => {
+    cy.swallowResizeObserverError();
+
+    cy.fixture<Regionaal>(`${vrcode}.json`)
+      .as('regionData')
+      .visit(`/veiligheidsregio/${vrcode}/${pageName}`);
+
+    cy.checkHeadings();
+  }
+);
+
+Cypress.Commands.add('swallowResizeObserverError', () => {
+  Cypress.on('uncaught:exception', (err, _runnable) => {
+    // For some reason this error throws very often during cypress tests,
+    // it doesn't crash anything, so for now we're just going to swallow
+    // the error and continue testing...
+    const errorMessage = err.toString();
+    if (errorMessage.includes('ResizeObserver loop')) {
+      return false;
+    }
+    return true;
+  });
+});
+
+Cypress.Commands.add(
+  'checkKpiValues',
+  (kpiTestInfo: Record<string, string | string[]>) => {
+    Object.entries(kpiTestInfo).forEach(([key, value]) => {
+      const element = cy.dataCy(key);
+      if (Array.isArray(value)) {
+        value.forEach((val) => {
+          element.contains(val);
+        });
+      } else {
+        element.contains(value);
+      }
```

Comment:
nit, but I think this can be written a bit simpler: 
```suggestion
      if (Array.isArray(value)) {
        [...value].forEach((val) => element.contains(val));
      }
```

Reply:
I'm not sure I follow your suggestion. you copy the array and then iterate? For now, I only removed the parentheses on the lambda, if that's the gist of what you mean.

Reply:
I think the point was to first convert whatever you have to an array (also if it's a single value) and then you can perform the same operation without needing the else statement to do something else.

I wonder if this works though. If value is a number for example wouldn't [...value] then throw, or is JS smart enough to convert it to a single value array?

Reply:
Ah yes that doesn't work .. my bad, and also the suggestion was buggy as hell. Guess I shouldn't do codereviews around 20:30.

What I actually had in mind was the following:

```tsx
[].concat(value).forEach(x => element.contains(val))
```

This will create an array and concat the `value` whether it's a string or an array of string:
```tsx
[].concat('plain string', ['string inside an array']) === ["plain string", "string inside an array"]